src/components/four-up.tsx

```diff
@@ -63,18 +66,27 @@ export class FourUpComponent extends BaseComponent<IProps, IState> {
   }
 
   public componentDidMount() {
-    if (this.container) {
-      this.grid.update({
-        height: this.container.offsetHeight - BORDER_SIZE,
-        initSplitters: true,
-        width: this.container.offsetWidth,
-      });
-    }
-    window.addEventListener("resize", this.handleResizeWindow);
+    this.resizeObserver = new ResizeObserver(entries => {
+      for (const entry of entries) {
+        if (entry.target === this.container) {
+          const {width, height} = entry.contentRect;
+          if (width > 0 && height > 0) {
+            this.grid.update({
+              height: height - BORDER_SIZE,
+              initSplitters: !this.roIsInitialized,
+              resizeSplitters: this.roIsInitialized,
+              width
+            });
+            this.roIsInitialized = true;
+          }
+        }
+      }
+    });
+    this.container && this.resizeObserver.observe(this.container);
```

Comment:
Much better approach.  I think there is a bunch more refactoring that would be useful in this module (there's a bunch of redundant code), but that is for another day!